identity/src/keypair.rs

```diff
@@ -724,4 +747,73 @@ mod tests {
 
         assert_implements_ord::<PublicKey>();
     }
+
+    #[test]
+    #[cfg(feature = "ed25519")]
+    fn test_publickey_from_ed25519_public_key_roundtrip_bytes() {
+        let keypair = Keypair::generate_ed25519();
+        let expected_keypair = keypair
+            .clone()
+            .try_into_ed25519()
+            .expect("A ed25519 keypair");
+        let expected_keypair_public_bytes = expected_keypair.public().to_bytes().to_vec();
+
+        let pub_key = ed25519::PublicKey::try_from_bytes(&expected_keypair_public_bytes)
+            .expect("A ed25519 public key");
+
+        let keypair_pub_key = PublicKey::from(pub_key);
+
+        assert_eq!(
+            keypair_pub_key.encode_protobuf(),
+            keypair.public().encode_protobuf()
+        );
```

Comment:
I am thinking this:


```suggestion
        let pubkey = Keypair::generate_ed25519().public();
        let ed25519_pubkey = pubkey
            .clone()
            .try_into_ed25519()
            .expect("A ed25519 public key);

        let converted_pubkey = PublicKey::from(ed25519_pubkey);

        assert_eq!(
            converted_pubkey,
            pubkey
        );
```

(Not sure if we implement `PartialEq` for pubkeys)

Reply:
Yes that works, changed it